cmd/webhook.go

```diff
@@ -233,29 +236,48 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
 			// Should deep copy because things change
 			tempSidecarConfig, _ := deepcopy.Anything(sidecarConfigTemplate)
 			sidecarConfig := tempSidecarConfig.(*Config)
-			bucketName := string(secretList.Items[sec].Data["S3_BUCKET"])
-			sidecarConfig.Containers[0].Name = filerName + "-" + bucketName + "-bucket-containers"
+
+			// Bucket might be a full path with shares, meaning with slashes (path1/path2)
+			bucketMount := string(secretList.Items[sec].Data["S3_BUCKET"])
+
+			// Take last path of share and limit length to 15 characters to manage length (max 63 for container names).
+			// The 15 characters limit was chosen arbitrarily and could be revised
+			bucketNamePaths := strings.Split(bucketMount, "/")
```

Comment:
In a later issue:
```
for str in bucketNamePaths {
 validate via regex
}
```

Reply:
later issue https://jirab.statcan.ca/browse/BTIS-418